src/js/events/SystemLogActions.js

```diff
@@ -0,0 +1,118 @@
+import {
+  REQUEST_SYSTEM_LOG_SUCCESS,
+  REQUEST_SYSTEM_LOG_ERROR,
+  REQUEST_PREVIOUS_SYSTEM_LOG_SUCCESS,
+  REQUEST_PREVIOUS_SYSTEM_LOG_ERROR
+} from '../constants/ActionTypes';
+import AppDispatcher from './AppDispatcher';
+import CookieUtils from '../utils/CookieUtils';
+import SystemLogUtil from '../utils/SystemLogUtil';
+
+/**
+ * Implementation of server sent event handling for
+ * https://github.com/dcos/dcos-log
+ */
+
+// Store of current open connections
+let sources = {};
+const SystemLogActions = {
+  subscribe(nodeID, options = {}) {
+    let {
+      subscriptionID
+    } = options;
+
+    let url = SystemLogUtil.getUrl(nodeID, options);
+    subscriptionID = subscriptionID || global.btoa(url);
+    let source = new EventSource(url, {
+      withCredentials: Boolean(CookieUtils.getUserMetadata())
+    });
+
+    source.addEventListener('message', function ({data, origin, eventPhase} = {}) {
+      if (origin !== global.location.origin && eventPhase !== EventSource.OPEN) {
+        // Event is not from same origin, or is not open anymore
+        return false;
+      }
+
+      let parsedData = JSON.parse(data);
```

Comment:
Let's wrap this in try/cate block if there is a chance that the received data is malformed?